db/migrate/20151011151130_create_course_user_achievements.rb

```diff
@@ -0,0 +1,13 @@
+class CreateCourseUserAchievements < ActiveRecord::Migration
+  def change
+    create_table :course_user_achievements do |t|
+      t.references :course_user
+      t.references :achievement, foreign_key: { references: :course_achievements }
+      t.datetime :obtained_at, null: false
```

Comment:
We sort the leaderboard by their achievements counts first, then obtained_at. obtained_at is the date when student submits his assessment.